graphql_compiler/query_formatting/cypher_formatting.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+import datetime
+import json
+from string import Template
+
+import arrow
+from graphql import GraphQLBoolean, GraphQLFloat, GraphQLID, GraphQLInt, GraphQLList, GraphQLString
+import six
+
+from ..compiler import CYPHER_LANGUAGE
+from ..compiler.helpers import strip_non_null_from_type
+from ..exceptions import GraphQLInvalidArgumentError
+from ..schema import GraphQLDate, GraphQLDateTime, GraphQLDecimal
+from .representations import represent_float_as_str, type_check_and_str
+
+
+def _safe_cypher_string(argument_value):
+    """Sanitize and represent a string argument in Cypher."""
+    if not isinstance(argument_value, six.string_types):
+        if isinstance(argument_value, bytes):  # likely to only happen in py2
+            argument_value = argument_value.decode('utf-8')
+        else:
+            raise GraphQLInvalidArgumentError(u'Attempting to convert a non-string into a string: '
+                                              u'{}'.format(argument_value))
+
+    # Using JSON encoding means that all unicode literals and special chars
+    # (e.g. newlines and backslashes) are replaced by appropriate escape sequences.
+    # Since the compiler only allows variables with upper or lower case letters (A-Z, a-z)
```

Comment:
Correct me if I am wrong, but I think that this is true for the name of the variable. Not for the value of the variable. So I don't see how this is relevant.

Reply:
I think I might've mixed up parameter names and parameter values when I wrote that comment.

For parameter names, it's enforced in helpers.py in `safe_variable_string`